core/arch/arm/sm/sm_a32.S

```diff
@@ -20,32 +20,39 @@
 
 	.section .text.sm_asm
 
+	.macro str_r2_sp_lr
+	str	r2, [r0], #4
+	str	sp, [r0], #4
+	str	lr, [r0], #4
+	.endm
+
 FUNC sm_save_unbanked_regs , :
 UNWIND(	.fnstart)
 UNWIND(	.cantunwind)
 	/* User mode registers has to be saved from system mode */
 	cps	#CPSR_MODE_SYS
-	stm	r0!, {sp, lr}
+	str	sp, [r0], #4
+	str	lr, [r0], #4
 
 	cps	#CPSR_MODE_IRQ
 	mrs	r2, spsr
-	stm	r0!, {r2, sp, lr}
+	str_r2_sp_lr
```

Comment:
I'd rather open code what's done in `str_r2_sp_lr` or define a macro that does all the stuff for a specified mode
Same goes for `ldr_r2_sp_lr` below.

Reply:
I'll introduce macros (`save_regs mode` and `restore_regs mode`).